### PR TITLE
Add unit test for `get_match_scores`

### LINKED ISSUE #35: Add unit test: match_scores: `get_match_scores`
The function `get_match_scores` does not have a unit test. It should have one.

### PR DESCRIPTION
**Pull request by mary-cleaton**
## Description

Use mocking to add a new unit test for `get_match_scores`, since this calls two child functions which each already have a unit test. Whilst doing this, also tweaked `test_assign_weights` so that its inputs and outputs are defined in a new `conftest.py` scripts and can be called by both `test_assign_weights` and `test_get_match_scores`.

closes #35 

## Type of change

*You can delete options that are not relevant.*

- [ ] Bug fix - *non-breaking change*
- [x] New feature - *non-breaking change*
- [ ] Breaking change - *backwards incompatible change, changes expected behaviour*
- [ ] Non-user facing change, structural change, dev functionality, docs ...

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented my code appropriately, focusing on explaining my design decisions (explain why, not how).
- [x] I have made corresponding changes to the documentation (comments, docstring, etc.. )
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have updated the change log.

<br>

#  Peer review
Any new code includes all the following:

- **Documentation**: docstrings, comments have been added/ updated.
- **Style guidelines**: New code conforms to the project's contribution guidelines.
- **Functionality**: The code works as expected, handles expected edge cases and exceptions are handled appropriately.
- **Complexity**: The code is not overly complex, logic has been split into appropriately sized functions, etc..
- **Test coverage**: Unit tests cover essential functions for a reasonable range of inputs and conditions. Added and existing tests pass on my machine.

### Review comments
Suggestions should be tailored to the code that you are reviewing. Provide context.
Be critical and clear, but not mean. Ask questions and set actions.
<details><summary>These might include:</summary>

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented
  - Do the tests effectively assure that it
  works correctly? Are there additional edge cases/ negative tests to be considered?
- code style improvements (could the code be written more clearly?)
</details>
<br>

*Further reading: [code review best practices](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html)*

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [semantic versioning](https://semver.org/spec/v2.0.0
   - `create_spark_session` in `tests/utils/test_utils_create_spark_session.py`.
   - `get_input_variables` in `tests/utils/utils.py`.
   - `get_deltas` in `tests/indicator_matrix/test_indicator_matrix.py`.
+  - `get_match_scores` in `test/match_scores/match_scores.py`.
```
Comment (danasebo) [inline CHANGELOG.md]: please correct `test/match_scores/match_scores.py` -> `tests/match_scores/test_match_scores.py` (correctly it should be `tests` and `test_match_scores.py`).

**Review by danasebo — CHANGES_REQUESTED** [review]
Looks good to me! No comments re functional changes / unit test as all work as expected (the test cases are sufficient, good use of conftest.py file, unit test run and provide 100% coverage).

The only very minor comment is a typo in the CHANGELOG.md file, once corrected this PR can be approved and merged.

File: CHANGELOG.md (hunk 1 above)
Comment (mary-cleaton) [reply CHANGELOG.md]: Done.

Comment (danasebo) [reply CHANGELOG.md]: please also correct "in test/m..." -> "in tests/m..."

Comment (mary-cleaton) [reply CHANGELOG.md]: Drat!
Fixed now.

**Review by danasebo — APPROVED** [review]
looks good to me
